google/resource_compute_instance_group_manager.go

```diff
@@ -178,47 +178,20 @@ func resourceComputeInstanceGroupManagerCreate(d *schema.ResourceData, meta inte
 		return err
 	}
 
-	targetSize := int64(0)
-	if v, ok := d.GetOk("target_size"); ok {
-		targetSize = int64(v.(int))
-	}
-
 	// Build the parameter
 	manager := &computeBeta.InstanceGroupManager{
-		Name:             d.Get("name").(string),
-		BaseInstanceName: d.Get("base_instance_name").(string),
-		InstanceTemplate: d.Get("instance_template").(string),
-		TargetSize:       targetSize,
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		BaseInstanceName:    d.Get("base_instance_name").(string),
+		InstanceTemplate:    d.Get("instance_template").(string),
+		TargetSize:          int64(d.Get("target_size").(int)),
+		NamedPorts:          getNamedPortsBeta(d.Get("named_port").([]interface{})),
+		TargetPools:         convertStringSet(d.Get("target_pools").(*schema.Set)),
+		AutoHealingPolicies: expandAutoHealingPolicies(d.Get("auto_healing_policies").([]interface{})),
 		// Force send TargetSize to allow a value of 0.
 		ForceSendFields: []string{"TargetSize"},
 	}
 
-	// Set optional fields
-	if v, ok := d.GetOk("description"); ok {
-		manager.Description = v.(string)
-	}
-
-	if v, ok := d.GetOk("named_port"); ok {
-		manager.NamedPorts = getNamedPortsBeta(v.([]interface{}))
-	}
-
-	if attr := d.Get("target_pools").(*schema.Set); attr.Len() > 0 {
-		var s []string
-		for _, v := range attr.List() {
-			s = append(s, v.(string))
-		}
-		manager.TargetPools = s
-	}
-
-	updateStrategy := d.Get("update_strategy").(string)
```

Comment:
Is it me or are we not setting `update_strategy` on create on the InstanceGroupManager? I don't see where this is being set (before and after your change)

Reply:
`update_strategy` is something we created. I was confused by that at first too. The `update_strategy` is used in the update method.

This field will go away as there is now a beta feature handling update for us. Once we replace our hacky client-side update mecanism by the more robust server-side update mecanish, we can remove it.